libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -495,6 +495,72 @@ void RC_Channel::init_aux_function(const aux_func_t ch_option, const aux_switch_
     }
 }
 
+#if !HAL_MINIMIZE_FEATURES
+
+const RC_Channel::LookupTable RC_Channel::lookuptable[] = {
+    { AUX_FUNC::SIMPLE_MODE,"SIMPLE_MODE"},
+    { AUX_FUNC::SAVE_TRIM,"SAVE_TRIM"},
+    { AUX_FUNC::SAVE_WP,"SAVE_WP"},
+    { AUX_FUNC::CAMERA_TRIGGER,"CAMERA_TRIGGER"},
+    { AUX_FUNC::RANGEFINDER,"RANGEFINDER"},
+    { AUX_FUNC::FENCE,"FENCE"},
+    { AUX_FUNC::RESETTOARMEDYAW,"RESETTOARMEDYAW"},
+    { AUX_FUNC::SUPERSIMPLE_MODE,"SUPERSIMPLE_MODE"},
+    { AUX_FUNC::SPRAYER,"SPRAYER"},
+    { AUX_FUNC::GRIPPER,"GRIPPER"},
+    { AUX_FUNC::PARACHUTE_ENABLE,"PARACHUTE_ENABLE"},
+    { AUX_FUNC::PARACHUTE_RELEASE,"PARACHUTE_RELEASE"},
+    { AUX_FUNC::PARACHUTE_3POS,"PARACHUTE_3POS"},
+    { AUX_FUNC::MISSION_RESET,"MISSION_RESET"},
+    { AUX_FUNC::RETRACT_MOUNT,"RETRACT_MOUNT"},
+    { AUX_FUNC::RELAY,"RELAY"},
+    { AUX_FUNC::LANDING_GEAR,"LANDING_GEAR"},
+    { AUX_FUNC::MOTOR_INTERLOCK,"MOTOR_INTERLOCK"},
+    { AUX_FUNC::RELAY2,"RELAY2"},
+    { AUX_FUNC::RELAY3,"RELAY3"},
+    { AUX_FUNC::RELAY4,"RELAY4"},
+    { AUX_FUNC::AVOID_ADSB,"AVOID_ADSB"},
+    { AUX_FUNC::PRECISION_LOITER,"PRECISION_LOITER"},
+    { AUX_FUNC::AVOID_PROXIMITY,"AVOID_PROXIMITY"},
+    { AUX_FUNC::WINCH_ENABLE,"WINCH_ENABLE"},
+    { AUX_FUNC::WINCH_CONTROL,"WINCH_CONTROL"},
+    { AUX_FUNC::RC_OVERRIDE_ENABLE,"RC_OVERRIDE_ENABLE"},
+    { AUX_FUNC::USER_FUNC1,"USER_FUNC1"},
+    { AUX_FUNC::USER_FUNC2,"USER_FUNC2"},
+    { AUX_FUNC::USER_FUNC3,"USER_FUNC3"},
+    { AUX_FUNC::LEARN_CRUISE,"LEARN_CRUISE"},
+    { AUX_FUNC::CLEAR_WP,"CLEAR_WP"},
+    { AUX_FUNC::SIMPLE,"SIMPLE"},
+    { AUX_FUNC::ZIGZAG_SaveWP,"ZIGZAG_SaveWP"},
+    { AUX_FUNC::COMPASS_LEARN,"COMPASS_LEARN"},
+    { AUX_FUNC::SAILBOAT_TACK,"SAILBOAT_TACK"},
+    { AUX_FUNC::GPS_DISABLE,"GPS_DISABLE"},
+    { AUX_FUNC::RELAY5,"RELAY5"},
+    { AUX_FUNC::RELAY6,"RELAY6"},
+    { AUX_FUNC::SAILBOAT_MOTOR_3POS,"SAILBOAT_MOTOR_3POS"},
+    { AUX_FUNC::SURFACE_TRACKING,"SURFACE_TRACKING"},
+    { AUX_FUNC::RUNCAM_CONTROL,"RUNCAM_CONTROL"},
+    { AUX_FUNC::RUNCAM_OSD_CONTROL,"RUNCAM_OSD_CONTROL"},
+    { AUX_FUNC::VISODOM_CALIBRATE,"VISODOM_CALIBRATE"},
+    { AUX_FUNC::Q_ASSIST ,"Q_ASSIST"},
+    { AUX_FUNC::CAM_MODE_TOGGLE,"CAM_MODE_TOGGLE"},
+    { AUX_FUNC::MAINSAIL, "MAINSAIL"},
+    { AUX_FUNC::FLAP,"FLAP"}
+};
+
+/* lookup the announcement for switch change */
+const char *RC_Channel::string_for_aux_function(AUX_FUNC function) const     
+{
+     for (const struct LookupTable entry : lookuptable) {
+        if (entry.option == function) {
+            return entry.announcement;
+        }
+     }
+     return nullptr;
```

Comment:
Don't you want to return "UNKNOWN" here?

Reply:
Nevermind, missed what you were going for.

Reply:
we don't want to be spitting out UNKNOWN for all the others, especially ones that are actually continuos (eg. fwd throttle on tilt rotors)